pkg/build/controller/controller.go

```diff
@@ -288,6 +290,23 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 	case kapi.PodRunning:
 		// The pod's still running
 		nextStatus = buildapi.BuildPhaseRunning
+	case kapi.PodPending:
+		if build.Status.Phase == buildapi.BuildPhasePending {
+			// When build is in pending state due to missing pull secrets for the image
+			// indicate this condition in reason.
+			if build.Status.Reason == buildapi.StatusReasonMissingPushSecret {
+				return nil
+			}
+			if secret := build.Spec.Output.PushSecret; secret != nil {
+				if bc.SecretClient.Secrets(build.Namespace).Get(secret.Name); err != nil && errors.IsNotFound(err) {
+					build.Status.Reason = buildapi.StatusReasonMissingPushSecret
+					if err := bc.BuildUpdater.Update(build.Namespace, build); err != nil {
```

Comment:
isn't this going to keep the build from getting updated to the "pending" state if the push secret is missing?  it's going to stay in "new" even though the pod is "pending"